### PR TITLE
feat: Implement the generated UI code structure proposal (react) - FUI-2139

### PR DESCRIPTION
**Pull request by ArturKrasinski**
[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

📓 &nbsp; **Related JIRA**

[FUI-2139](https://genesisglobal.atlassian.net/browse/FUI-2139)

🤔 &nbsp; **What does this PR do?**

- Implement the generated UI code structure proposal

🚀 &nbsp; **Where should the reviewer start?**

- .genx/templates/react/component/*
- .genx/utils/generateTile.js

📑 &nbsp; **How should this be tested?**

### Defaults (react)

```
rm -rf blankappseedtest && npx -y @genesislcap/genx@latest init blankappseedtest -x --ref ak/FUI-2139 --framework React --no-npm
```

### Route and CSV parameter handling test (react)

```
rm -rf blankappseedtest && npx -y @genesislcap/genx@latest init blankappseedtest --ref ak/FUI-2139 --framework React  --no-npm -x --routes '[{"name":"Home","tiles":[{"title":"Entity manager","type":"entity-manager","config":{ "modalPosition": "centre", "sizeColumnsToFit": true, "enableSearchBar": true, "resourceName":"ALL_POSITIONS","title":"My Positions","updateEvent":"EVENT_COUNTERPARTY_MODIFY","deleteEvent":"EVENT_COUNTERPARTY_DELETE","createEvent":"EVENT_COUNTERPARTY_INSERT", "createFormUiSchema": {"type":"VerticalLayout","elements":[{"type":"Control","label":"Inline create schema - main contact","scope":"#/properties/MAIN_CONTACT"},{"type":"Control","label":"Issuer Name - Local Schema","scope":"#/properties/ISSUER_NAME","options":{"readonly":true}},{"type":"Control","label":"Price","scope":"#/properties/PRICE"},{"type":"Control","scope":"#/properties/COUNTERPARTY","options":{"allOptionsResourceName":"ALL_COUNTERPARTYS","valueField":"COUNTERPARTY_ID","labelField":"COUNTERPARTY_ID"}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"isPassword":true}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"textarea":true}}]}, "updateFormUiSchema": {"type":"VerticalLayout","elements":[{"type":"Control","label":"Inline update schema - main contact","scope":"#/properties/MAIN_CONTACT"},{"type":"Control","label":"Issuer Name - Local Schema","scope":"#/properties/ISSUER_NAME","options":{"readonly":true}},{"type":"Control","label":"Price","scope":"#/properties/PRICE"},{"type":"Control","scope":"#/properties/COUNTERPARTY","options":{"allOptionsResourceName":"ALL_COUNTERPARTYS","valueField":"COUNTERPARTY_ID","labelField":"COUNTERPARTY_ID"}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"isPassword":true}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"textarea":true}}]}, "columns": [{"field":"INSTRUMENT_NAME","headerName":"Instrument Name"},{"field":"VALUE","headerName":"Inline coldef - VALUE"},{"field":"QUANTITY","headerName":"Quantity"},{"field":"PNL","headerName":"PNL"}] }},{"title":"Grid","type":"grid-pro","config":{"resourceName":"ALL_TRADES", "gridOptions": {"columnDefs":[{"field":"INSTRUMENT_NAME","headerName":"Instrument Name"},{"field":"VALUE","headerName":"Inline coldef - VALUE"},{"field":"QUANTITY","headerName":"Quantity"},{"field":"PNL","headerName":"PNL"}]} }},{"title":"Form","type":"smart-form","config":{"resourceName":"EVENT_COUNTERPARTY_INSERT", "uischema": {"type":"VerticalLayout","elements":[{"type":"Control","label":"Inline form schema - main contact","scope":"#/properties/MAIN_CONTACT"},{"type":"Control","label":"Issuer Name - Local Schema","scope":"#/properties/ISSUER_NAME","options":{"readonly":true}},{"type":"Control","label":"Price","scope":"#/properties/PRICE"},{"type":"Control","scope":"#/properties/COUNTERPARTY","options":{"allOptionsResourceName":"ALL_COUNTERPARTYS","valueField":"COUNTERPARTY_ID","labelField":"COUNTERPARTY_ID"}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"isPassword":true}},{"type":"Control","label":"Password","scope":"#/properties/PASSWORD","options":{"textarea":true}}]} }}]}, {"name":"Realtime Dashboard","tiles":[{"title":"Entity manager tile","type":"entity-manager","config":{"resourceName":"ALL_COUNTERPARTYS","title":"Counterparty Management","updateEvent":"EVENT_COUNTERPARTY_MODIFY","deleteEvent":"EVENT_COUNTERPARTY_DELETE","createEvent":"EVENT_COUNTERPARTY_INSERT"}},{"title":"Form tile","type":"smart-form","config":{"resourceName":"EVENT_COUNTERPARTY_INSERT"}}]},{"name":"Analytics","tiles":[{"title":"Grid Tile","type":"grid-pro","config":{"resourceName":"ALL_POSITIONS"}},{"title":"Charts Tile 1","type":"chart","config":{"type":"line","resourceName":"ALL_POSITIONS","xField":"INSTRUMENT_NAME","yField":"VALUE"}},{"title":"Charts Tile 2","type":"chart","config":{"type":"pie","resourceName":"ALL_POSITIONS","xField":"INSTRUMENT_NAME","yField":"VALUE"}},{"title":"Charts Tile 3","type":"chart","config":{"type":"column","resourceName":"ALL_POSITIONS","xField":"INSTRUMENT_ID","yField":"VALUE"}}]}]' --csv '[{"name": "trade", "fields": ["a", "B"]}, {"name": "position", "fields": ["id", "TYPE"]} ]' --apiHost 'wss://public-foundation.genesislab.global/gwf/' --no-shell && cd blankappseedtest/client && npm run bootstrap && npm run dev
```

✅ &nbsp; **Checklist**

- [x] I have tested my changes.
- [ ] I have added tests for my changes.
- [ ] I have updated the project documentation to reflect my changes.

### CODE REVIEW

File: .genx/templates/react/component/component.column.defs.hbs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ColDef } from '@ag-grid-community/core';
+import { getNumberFormatter, getDateFormatter } from '@genesislcap/foundation-utils';
+{{#if route.FDC3EventHandlersEnabled}}
+    import { sendEventOnChannel } from '../../../utils';
```
Comment (patrickoneill-genesis) [inline .genx/templates/react/component/component.column.defs.hbs:4]: nitpick - should the indentation be removed here?

File: .genx/templates/react/component/component.gridOptions.hbs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { GridOptionsConfig } from '@genesislcap/rapid-grid-pro';
+import { getNumberFormatter, getDateFormatter } from '@genesislcap/foundation-utils';
+{{#if route.FDC3EventHandlersEnabled}}
+    import { sendEventOnChannel } from '../../../utils';
```
Comment (patrickoneill-genesis) [inline .genx/templates/react/component/component.gridOptions.hbs:4]: nitpick - indentation

Comment (ArturKrasinski) [reply .genx/templates/react/component/component.gridOptions.hbs]: removed

File: .genx/templates/react/component/component.column.defs.hbs (hunk 1 above)
Comment (ArturKrasinski) [reply .genx/templates/react/component/component.column.defs.hbs]: removed

**Comment by skawian** [comment]
tested locally and works fine, created [FUI-2153](https://genesisglobal.atlassian.net/browse/FUI-2153) and [FUI-2154](https://genesisglobal.atlassian.net/browse/FUI-2154) as follow up items

**Review by skawian — APPROVED** [review]

File: .genx/templates/react/component/component.hbs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { Component, Input, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { getUser } from '@genesislcap/foundation-user';
+import { ErrorMessageComponent } from '../../../components/error-message/error-message.component';
+import { getViewUpdateRightComponent } from '../../../utils';
+{{#if tile.config.gridOptions}}
+import { GridOptionsConfig } from "@genesislcap/rapid-grid-pro";
```
Comment (patrickoneill-genesis) [inline .genx/templates/react/component/component.hbs:7]: nitpick - single quotes.

Comment (patrickoneill-genesis) [reply .genx/templates/react/component/component.hbs:7]: as good practice should this be with the relative imports

**Pull request merged**